hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/FactoryIdHelper.java

```diff
@@ -18,154 +18,94 @@
+        MANAGEMENT_DS("hazelcast.serialization.ds.management", -24),
+        TEXT_PROTOCOLS_DS("hazelcast.serialization.ds.text.protocols", -25),
+        ENTERPRISE_HOTRESTART_BACKUP_DS("hazelcast.serialization.ds.spi.hotrestart.backup", -26),
+        SCHEDULED_EXECUTOR_DS("hazelcast.serialization.ds.scheduled.executor", -27),
+        USER_CODE_DEPLOYMENT_DS("hazelcast.serialization.ds.user.code.deployment", -28),
+        AGGREGATOR_DS("hazelcast.serialization.ds.aggregator", -29),
+        PROJECTION_DS("hazelcast.serialization.ds.projection", -30),
+        CONFIG_DS("hazelcast.serialization.ds.config", -31),
+        ENTERPRISE_SECURITY_DS("hazelcast.serialization.ds.security", -32),
+        EVENT_JOURNAL_DS("hazelcast.serialization.ds.event_journal", -33),
+        FLAKE_ID_GENERATOR_DS("hazelcast.serialization.ds.flake_id_generator", -34),
+        SPLIT_BRAIN_DS("hazelcast.serialization.ds.split_brain", -35),
+        PN_COUNTER_DS("hazelcast.serialization.ds.pn_counter", -36),
+        METRICS_DS("hazelcast.serialization.metrics", -37),
+        SQL_DS("hazelcast.serialization.sql", -38),
+        JSON_DS("hazelcast.serialization.json", -39),
+        UTIL_COLLECTION_DS("hazelcast.serialization.util.collection", -40),
+        JET_SQL_DS("hazelcast.serialization.jet.sql", -41),
+        SCHEMA_DS("hazelcast.serialization.schema", -42),
+        ENTERPRISE_PARTITION_DS("hazelcast.serialization.ds.enterprise.partition", -43),
+        BASIC_FUNCTIONS_DS("hazelcast.serialization.lambda", -44);
+
+        private final String prop;
+        /** factory ID 0 is reserved for Cluster objects (Data, Address, Member etc)... */
+        private final int defaultFactoryId;
+
+        Factory(final String prop, final int defaultFactoryId) {
+            this.prop = prop;
+            this.defaultFactoryId = defaultFactoryId;
+        }
 
-    public static final String SCHEMA_DS_FACTORY = "hazelcast.serialization.schema";
-    public static final int SCHEMA_DS_FACTORY_ID = -42;
+        Factory(final String prop) {
+            this(prop, 0);
+        }
 
-    public static final String ENTERPRISE_PARTITION_DS_FACTORY = "hazelcast.serialization.ds.enterprise.partition";
-    public static final int ENTERPRISE_PARTITION_DS_FACTORY_ID = -43;
+        public int getDefaultFactoryId() {
+            return defaultFactoryId;
+        }
 
-    public static final String BASIC_FUNCTIONS_DS_FACTORY = "hazelcast.serialization.lambda";
-    public static final int BASIC_FUNCTIONS_DS_FACTORY_ID = -44;
+        public int getFactoryId() {
+            return FactoryIdHelper.getFactoryId(prop, defaultFactoryId);
+        }
 
-    // factory ID 0 is reserved for Cluster objects (Data, Address, Member etc)...
+        /** @return the {@link toString} of the {@link Factory} indexed by {@code factoryId}, else the {@code factoryId} */
+        public static String getName(final int factoryId) {
```

Comment:
> My spider senses are tingling here.
> 
> I'm unsure how frequently the getName is being called. But if this is done frequent, then I don't know how the following code is going to behave.
> 
> Before merging this code, we need to make sure we do not run into a performance regression because we want to write idiomatic Java code.

This is only called when a `HazelcastSerializationException` is thrown. At that point I think the value of getting a better message is more important than performance.

Reply:
If it is only during an exception it is less of an issue since an Exception is already very expensive.